### DISCUSSION TITLE
Pages Routes Support

### DISCUSSION THREAD

**Discussion by eroldmrclk** [vercel/next.js#50953] [App Router] [answered]
With the release of version 13, Next.js achieved stability in terms of app routes. However, can we expect consistent support for pages routes in future versions? Also, is there a specified end-of-support date for pages routes?

**Comment by icyJoseph** [comment, marked as answer]
From the announcement blog post:

- https://nextjs.org/blog/next-13-4#is-the-pages-router-going-away

> [Is the Pages Router going away?](https://nextjs.org/blog/next-13-4#is-the-pages-router-going-away)
> No. We are committed to supporting pages/ development, including bug fixes, improvements, and security patches, for multiple major versions moving forward. We want to ensure developers have enough time to incrementally adopt the App Router as they're ready.
> 
> Using both pages/ and app/ in production together is supported and encouraged. The App Router can be adopted on a [per-route basis](https://nextjs.org/docs/app/building-your-application/upgrading/app-router-migration).


Notice that it says, multiple major versions. I don't think there'll ever be a hard line out of the blue, they'll handle this proper when the time comes, if ever. Pages router is a solid piece of tech that has developed over several years, it just won't go up in smoke all of the sudden.
